pkg/provider/provider.go

```diff
@@ -222,3 +281,55 @@ func ParsePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
 	}
 	return rsaPrivateKey, nil
 }
+
+type Result struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"`
+}
+
+func GetOauthData(refreshToken, redirectUrl string) url.Values {
+	data := url.Values{}
+	data.Set("grant_type", "refresh_token")
+	data.Set("refresh_token", refreshToken)
+	data.Set("redirect_uri", redirectUrl)
+	return data
+}
+
+func GetOauthRequest(dataContent *strings.Reader, endPoint, clientId, clientSecret string) *http.Request {
+	request, err := http.NewRequest("POST", endPoint, dataContent)
+	if err != nil {
+		errors.Wrap(err, "Request to the endpoint could not be completed")
```

Comment:
```suggestion
		errors.Wrap(err, "Request to the endpoint could not be completed")
		return err
```

Reply:
qq? the return type in this is a http.Request, you want me to return nil and update the error

Reply:
I believe we cant just return like that, because the return type should be an error and we are returning a htp.Request.

Reply:
`		return nil, errors.Wrap(err, "Request to the endpoint could not be completed")`

Reply:
i believe we cant just return something like that, i am assuming you guys want me to do this instead
`func GetOauthRequest(dataContent io.Reader, endPoint, clientId, clientSecret string) (*http.Request, error) {
	request, err := http.NewRequest("POST", endPoint, dataContent)
	if err != nil {
		return nil, errors.Wrap(err, "Request to the endpoint could not be completed")
	}
	request.SetBasicAuth(clientId, clientSecret)
	request.Header.Set("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
	return request, nil

}`